cachito/web/api_v1.py

```diff
@@ -33,14 +36,54 @@ def get_request(request_id):
     :return: a Flask JSON response
     :rtype: flask.Response
     :raise NotFound: if the request is not found
-    :raise ValidationError: if the request ID is of invalid format
     """
-    try:
-        request_id = int(request_id)
-    except ValueError:
-        raise ValidationError(f'{request_id} is not a valid request ID')
+    return flask.jsonify(Request.query.get_or_404(request_id).to_json())
+
+
+@api_v1.route('/requests/<int:request_id>/download', methods=['GET'])
+def download_archive(request_id):
+    """
+    Download archive of source code.
+
+    :param int request_id: the value of the request ID
+    :return: a Flask send_file response
+    :rtype: flask.Response
+    :raise NotFound: if the request is not found
+    """
     request = Request.query.get_or_404(request_id)
-    return flask.jsonify(request.to_json())
+    # TODO: Verify request has already been processed.
+
+    cachito_shared_dir = flask.current_app.config.get('CACHITO_SHARED_DIR')
```

Comment:
Some sort of validation should occur so that `cachito_shared_dir` is not `None`. I'm also fine if a KeyError is raised since it'd only happen due to a misconfiguration.

Reply:
Sticking to `KeyError` for now. 

NOTE: I think we should make the configuration a bit more user friendly. Since the main deployment target for cachito is a containerized environment, we could optimize for that. I have some thoughts on this, but not yet ready to share. I'm mentioning it here to give some reasoning why I think we should stick to `KeyError` for now :)